test_sanity.py

```diff
@@ -1,57 +1,14 @@
-import requests
-from requests.exceptions import JSONDecodeError
-
-def make_valid_payload(method: str, params: dict | None = None) -> dict:
-    payload = {"method": method, "jsonrpc": "2.0", "id": 1}
-    
-    if params:
-        payload["params"] = params
-
-    return payload
-
-
-def make_valid_request(method: str, params: dict | None = None) -> dict:
-    payload = make_valid_payload(method=method, params=params)
-    sensor_response = send_post(**payload)
-    return sensor_response.get("result", {})
-
-def send_post(method: str | None = None, params: dict | None = None, jsonrpc: str | None = None, id: int | None = None):
-    request_body = {}
-
-    if method:
-        request_body["method"] = method
-    
-    if params:
-        request_body["params"] = params
-    
-    if jsonrpc:
-        request_body["jsonrpc"] = jsonrpc
-    
-    if id:
-        request_body["id"] = id
-
-    request_headers = {"Authorization": "0000"}
-    print(request_body)
-    res = requests.post("http://127.0.0.1:9898/rpc", json=request_body, headers=request_headers)
-    
-    try:
-        print(res.status_code)
-        print(res.text)
-        return res.json()
-    except JSONDecodeError:
-        return {}
-
-
-def get_sensor_info():
-    return make_valid_request("get_info")
-
-
-def get_sensor_reading():
-    return make_valid_request("get_reading")
-
-
-def test_sanity():
+def test_sanity(get_sensor_info, get_sensor_reading, get_sensor_methods, set_sensor_name, set_reading_interval, reset_to_factory, update_firmware, reboot):
     sensor_info = get_sensor_info()
+    """"
```

Comment:
Since these are not used anywhere in the test, it would be best to remove them from here.

Reply:
updated test_sanity.py file (removed not needed data)